### ISSUE TITLE
Project creation does not happen on first chef-client

### ISSUE THREAD

**Issue by atheiman** [Webtrends/rundeck#136]
**tl;dr: Turns out rundeck isnt running yet when projects are trying to be created. The project creation logic needs to be moved out of rundeck::server and into a new recipe after rundeck::apache is run.**

Projects are currently not created on the first chef-client run (in test-kitchen at least, Im not sure about with a real node). ~~I think there is an issue with apache or rundeckd restarting too close to right before the `rd-project -p #{project} -a create` command is run in the server_install recipe.~~ edit: i debunked this below

also, it doesnt help that the project creation is guarded by
```ruby
    # will return 0 if grep matches
    # only run if project does not exist
    only_if "rd-jobs -p #{project} list 2>&1 | grep -q '^ERROR .*project does not exist'"
```

I assume `rd-jobs` is returning some other error (`rundeckd not running` perhaps) and project creation is being skipped.

I identified this bug by adding a localhost port forward from my mac into the vm (`8080 on host => 80 on guest`), logging in as the admin, and seeing no projects. After converging the node again, I login and see the localhost project. You can also see in the chef-client output from the initial run:

```
         * bash[check-project-localhost] action run (skipped due to only_if)
```

and from the second run

```
         * bash[check-project-localhost] action run
           - execute "bash"  "/tmp/chef-script20170128-5488-f89mr3"
```

When projects are managed via the api client, this should be resolved I think. If not, the issue will become more obvious (the api will respond with a 500).

**Comment by atheiman** [comment]
I've added the following into the server_install recipe just before the project create command would be run:

```ruby
  execute "rd-jobs -p #{project} list 2>&1" do
    live_stream true
  end
```

And the first chef-client results in

```
         * execute[rd-jobs -p localhost list 2>&1] action run
           [execute] Error: Error making server request to http://localhost:4440: Error occurred while trying to authenticate to server: Connection refused (Connection refused)

           ================================================================================
           Error executing action `run` on resource 'execute[rd-jobs -p localhost list 2>&1]'
           ================================================================================

           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           Expected process to exit with [0], but received '1'
           ---- Begin output of rd-jobs -p localhost list 2>&1 ----
           STDOUT: Error: Error making server request to http://localhost:4440: Error occurred while trying to authenticate to server: Connection refused (Connection refused)
           STDERR:
           ---- End output of rd-jobs -p localhost list 2>&1 ----
           Ran rd-jobs -p localhost list 2>&1 returned 1

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/rundeck/recipes/server_install.rb

           279:   execute "rd-jobs -p #{project} list 2>&1" do
           280:     live_stream true
           281:   end
           282:

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/rundeck/recipes/server_install.rb:279:in `block in from_file'

           execute("rd-jobs -p localhost list 2>&1") do
             action [:run]
             retries 0
             retry_delay 2
             default_guard_interpreter :execute
             command "rd-jobs -p localhost list 2>&1"
             backup 5
             returns 0
             live_stream true
             declared_type :execute
             cookbook_name "rundeck"
             recipe_name "server_install"
           end

           Platform:
           ---------
           x86_64-linux

         * service[rundeck] action nothing (skipped due to action :nothing)

       Running handlers:
       [2017-01-29T00:35:24+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2017-01-29T00:35:24+00:00] ERROR: Exception handlers complete
       Chef Client failed. 37 resources updated in 01 minutes 46 seconds
       [2017-01-29T00:35:24+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2017-01-29T00:35:24+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
       [2017-01-29T00:35:24+00:00] ERROR: execute[rd-jobs -p localhost list 2>&1] (rundeck::server_install line 279) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
       ---- Begin output of rd-jobs -p localhost list 2>&1 ----
       STDOUT: Error: Error making server request to http://localhost:4440: Error occurred while trying to authenticate to server: Connection refused (Connection refused)
       STDERR:
       ---- End output of rd-jobs -p localhost list 2>&1 ----
       Ran rd-jobs -p localhost list 2>&1 returned 1
       [2017-01-29T00:35:25+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

Running chef-client minutes later resulted in the same. Turns out rundeck isnt running yet when projects are trying to be created. this project logic needs to be moved out of rundeck::server and into a new recipe after rundeck::apache is run.
